lib/boom_notifier.ex

```diff
@@ -1,8 +1,39 @@
 defmodule BoomNotifier do
   @moduledoc false
+
   # Notify the exception to all the defined notifiers
 
   alias BoomNotifier.ErrorStorage
+  require Logger
+
+  def run_callback(settings, callback) do
+    missing_keys = Enum.reject([:notifier, :options], &Keyword.has_key?(settings, &1))
+
+    case missing_keys do
+      [] ->
+        callback.(settings[:notifier], settings[:options])
+
+      [missing_key] ->
+        Logger.error("Settings error: #{inspect(missing_key)} parameter missing")
+
+      _ ->
+        Logger.error(
+          "Settings error: The following parameters are missing: #{inspect(missing_keys)}"
+        )
+    end
+  end
+
+  def walkthrough_notifiers(settings, callback) do
+    case Keyword.get(settings, :notifiers) do
+      nil ->
+        run_callback(settings, callback)
+
+      notifiers_settings when is_list(notifiers_settings) ->
+        for notifier_settings <- notifiers_settings do
```

Comment:
I would expect to see `for or notifier_setting <- notifiers_settings do` (note the singular instead of the plural in the var at the left of the ->).

Using the same name works perhaps, but it seems super confusing. Another option, that is totally acceptable, is using a shorter name for the iterator: e.g.

```ex
for s <- notifiers_settings, do: run_callback(s, callback)
```

Reply:
Thinking a little more, you can also use `Enum.each`. Note the comprehension syntax (->) returns a list, but in this case, the resulting list is totally ignored. `Enum.each` simply returns `:ok` (because something has to be returned in Elixir in each expression, but it is more natural than returning a list of empty values)
```ex
Enum.each(notifiers_settings, &run_callback(&1, callback))
```

What do you think?